src/server.rs

```diff
@@ -244,14 +244,10 @@ fn read_opt(
     }
     slog_debug!(logger.log, "Found messages"; "len" => messages.len(), "user_id" => &user_id);
     // returns json {"status":200, "index": max_index, "messages":[{"index": #, "data": String}, ...]}
-    let is_last = match limit {
-        None => true,
-        Some(0) => true,
-        Some(_) => {
-            let last = messages.last().unwrap();
-            (last.idx as u64) == max_index
-        }
-    };
+    let is_last = messages
+        .last()
+        .map(|last| (last.idx as u64) == max_index)
+        .unwrap_or(true);
```

Comment:
This doesn't do the same thing as the above block it replaces, if it's possible that messages could be retrieved and the limit is set to None.